core/src/testUtils/groovy/nu/yona/server/YonaServer.groovy

```diff
@@ -172,4 +182,64 @@ class YonaServer
 	{
 		DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(dateTime)
 	}
+
+	static def relativeDateTimeStringToZonedDateTime(relativeDateTimeString)
+	{
+		def fields = relativeDateTimeString.tokenize(' ')
+		assert fields.size() <= 3
+		assert fields.size() > 0
+		int parsedFields = 0
+		int weekOffset = 0
+		int dayOffset = 0
+
+		LocalDateTime now = LocalDateTime.now()
+		switch (fields.size())
+		{
+			case 3:
+				assert fields[0].startsWith("W")
+				weekOffset = Integer.parseInt(fields[0].substring(1))
+				parsedFields++
+			// Fall through
+			case 2:
+				int weekDay = DateTimeFormatter.ofPattern("eee").parse(fields[parsedFields]).get(ChronoField.DAY_OF_WEEK)
```

Comment:
I made a mistake reading my calendar. It returns May 22.

If anything goes wrong, it will be when running the tests on a Sunday, but I'm not entirely sure of that.